src/metabase/api/user.clj

```diff
@@ -251,13 +250,10 @@
   [user]
   (let [coll-ids-filter (collection/visible-collection-ids->honeysql-filter-clause
                           :collection_id
-                          (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))
-        perms-query {:where [:and
-                             [:= :archived false]
-                             coll-ids-filter]}]
-    #_{:clj-kondo/ignore [:discouraged-var]}
-    (assoc user :has_question_and_dashboard (and (db/exists? 'Card (perms-query user))
-                                                 (db/exists? 'Dashboard (perms-query user))))))
+                          (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))]
+    (assoc user :has_question_and_dashboard (and coll-ids-filter
```

Comment:
`coll-ids-filter` is just honeySQL that needs to be included in the `Card` and `Dashboard` queries; it's not a query in and of itself